tests/NexusMods.DataModel.Tests/FileOriginRegistryTests.cs

```diff
@@ -0,0 +1,140 @@
+using System.Diagnostics.CodeAnalysis;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.DiskState;
+using NexusMods.Abstractions.FileExtractor;
+using NexusMods.Abstractions.FileStore;
+using NexusMods.Abstractions.FileStore.ArchiveMetadata;
+using NexusMods.Abstractions.Games.Downloads;
+using NexusMods.Abstractions.IO;
+using NexusMods.Abstractions.Serialization.Attributes;
+using NexusMods.Abstractions.Serialization.DataModel;
+using NexusMods.Abstractions.Serialization.DataModel.Ids;
+using NexusMods.DataModel.Tests.Harness;
+using NexusMods.Hashing.xxHash64;
+using NSubstitute;
+
+namespace NexusMods.DataModel.Tests;
+
+public class FileOriginRegistryTests : ADataModelTest<FileOriginRegistryTests>
+{
+    public FileOriginRegistryTests(IServiceProvider provider) : base(provider)
+    {
+
+    }
+
+    [Fact]
+    public async Task RegisterFolder_ShouldRegisterCorrectly()
```

Comment:
See if you want to add a test for checking against entries that are in the datastore without an associated `DownloadAnalysis`